pkg/cluster/nodeutils/util.go

```diff
@@ -126,3 +126,64 @@ func ImageID(n nodes.Node, image string) (string, error) {
 	}
 	return crictlOut.Status.ID, nil
 }
+
+// ImageTags is used to perform a reverse lookup of the ImageID to list set of available
+// RepoTags corresponding to the ImageID in question
+func ImageTags(n nodes.Node, imageID string) (map[string]bool, error) {
+	var out bytes.Buffer
+	tags := make(map[string]bool, 0)
+	if err := n.Command("crictl", "inspecti", imageID).SetStdout(&out).Run(); err != nil {
+		return tags, err
+	}
+	crictlOut := struct {
+		Status struct {
+			RepoTags []string `json:"repoTags"`
+		} `json:"status"`
+	}{}
+	if err := json.Unmarshal(out.Bytes(), &crictlOut); err != nil {
+		return tags, err
+	}
+	for _, tag := range crictlOut.Status.RepoTags {
+		tags[tag] = true
+	}
+	return tags, nil
+}
+
+// ReTagImage is used to tag an ImageID with a custom tag specified by imageName parameter
+func ReTagImage(n nodes.Node, imageID, imageName string) error {
+	var out bytes.Buffer
+	return n.Command("ctr", "--namespace=k8s.io", "images", "tag", "--force", imageID, imageName).SetStdout(&out).Run()
+}
+
+// sanitizeImage is a helper to return human readable image name and
+// the podman pullable image name from the provided image
+func SanitizeImage(image string) (friendlyImageName, pullImageName string) {
```

Comment:
This one probably shouldn't be made public, and probably shouldn't be shared between podman and containerd, even if the method _currently_ does what we need for both.

What we're _currently_ doing for images passed to podman we should probably not do for all images and consider not doing in the future, and just make people actually specify docker.io when they want to use kind images from there.

For containerd, you could imagine this behavior could also become configurable in the future there.

TDLR: WET > DRY
Write Everything Twice. We'll know it's a good abstraction or not by the third time :-)

Reply:
we need a much simpler version of this for containerd loading, which doesn't care about the sha part etc., and will be easier to maintain someday.

podman and docker already have different versions of producing the "friendly" image for printing, which was also possibly a mis-feature (as it obscures what exactly we're running ...)

Reply:
@BenTheElder All right. Let me put the refactoring back to the way it was and create a much simpler version of the function for sanitizing the image name for our current use-case. I will update the changes in a while